ydb/core/blobstorage/dsproxy/dsproxy_get_impl.cpp

```diff
@@ -313,6 +318,7 @@ void TGetImpl::PrepareVPuts(TLogContext &logCtx, TDeque<std::unique_ptr<TEvBlobS
         auto vput = std::make_unique<TEvBlobStorage::TEvVPut>(put.Id, put.Buffer, vdiskId, true, nullptr, Deadline,
             Blackboard.PutHandleClass);
         DSP_LOG_DEBUG_SX(logCtx, "BPG15", "Send put to orderNumber# " << put.OrderNumber << " vput# " << vput->ToString());
+        History.AddVPut(put.Id.PartId(), 1, put.OrderNumber);
```

Comment:
Consider moving AddVPut/AddVGet to the place where messages get sent (or processed somehow).